osu.Game.Rulesets.Osu/Mods/OsuModRelax.cs

```diff
@@ -101,7 +104,7 @@ public void Update(Playfield playfield)
 
             if (requiresHold)
                 changeState(true);
-            else if (isDownState && time - lastStateChangeTime > AutoGenerator.KEY_UP_DELAY)
+            else if ((isDownState && time + AutoGenerator.KEY_UP_DELAY >= beatmapLength) || time - lastStateChangeTime > AutoGenerator.KEY_UP_DELAY)
```

Comment:
What's with this condition grouping? So there will always be a release frame emitted if `time - lastStateChangeTime > KEY_UP_DELAY`, even if no action was pressed in the first place? That sounds unnecessary at best.

Reply:
> even if no action was pressed in the first place? 

It triggers when
Both `isDownState and BeatmapEnds` 
or 
`time - lastStateChangeTime > AutoGenerator.KEY_UP_DELAY`

Reply:
Should be more proper with https://github.com/ppy/osu/pull/13891#discussion_r671434239 now.